File: JavaScript/53. Maximum Subarray.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var maxSubArray = function(nums) {
    var curMax = 0;
    var maxTillNow = Number.MIN_SAFE_INTEGER;
    for(var c of nums)
        curMax = Math.max(c, curMax + c),
        maxTillNow = Math.max(maxTillNow, curMax);
    return maxTillNow;
};

File: Python/53. Maximum Subarray.py
class Solution:
    def maxSubArray(self, nums: List[int]) -> int:
        curMax = 0
        maxTillNow = float('-inf')
        for c in nums:
            curMax = max(c, curMax + c)
            maxTillNow = max(maxTillNow, curMax)
        return maxTillNow
